cmd/mailchain/internal/settings/mailbox_state.go

```diff
@@ -1,44 +1,60 @@
 package settings
 
 import (
+	"fmt"
+
 	"github.com/mailchain/mailchain/cmd/mailchain/internal/settings/defaults"
 	"github.com/mailchain/mailchain/cmd/mailchain/internal/settings/output"
 	"github.com/mailchain/mailchain/cmd/mailchain/internal/settings/values"
 	"github.com/mailchain/mailchain/stores"
+	"github.com/mailchain/mailchain/stores/bdbstore"
 	"github.com/mailchain/mailchain/stores/ldbstore"
-	"github.com/pkg/errors"
 )
 
+// State interface wraps basic methods required for mailbox state.
+type State interface {
+	Produce() (stores.State, error)
+	Output() output.Element
+	Kind() values.String
+}
+
+// mailboxState returns the default MailboxState which uses leveldb store.
 func mailboxState(s values.Store) *MailboxState {
-	k := &MailboxState{
-		Kind:                values.NewDefaultString(defaults.MailboxStateKind, s, "mailboxState.kind"),
-		mailboxStateLevelDB: mailboxStateLevelDB(s),
+	return &MailboxState{
+		mailboxState: mailboxStateLevelDB(s),
+	}
+}
+
+// mailboxStateFromKind returns MailboxState based on store kind.
+func mailboxStateFromKind(kind string, s values.Store) (*MailboxState, error) {
+	switch kind {
+	case StoreLevelDB:
+		m := &MailboxState{mailboxState: mailboxStateLevelDB(s)}
+		return m, nil
+	case StoreBadgerDB:
+		m := &MailboxState{mailboxState: mailboxStateBadgerDB(s)}
+		return m, nil
+	default:
+		return nil, fmt.Errorf("unknown state store %q", kind)
 	}
-	return k
 }
```

Comment:
This function is not required as it will not be called by the desired implementation. The logic of it though is correct and will instead be spread over a couple of different functions.